README.md

```diff
@@ -31,8 +31,99 @@ transitive dependencies:
 `models`
 `testing` -> `api` -> `models`
 
-#### Adding Tests
-Tests are located in the `Tests/` directory. To add a new test, either write a new function in one of the existing classes, or create a new class and add the function into the `Tests/` directory.
+#### Writing Tests
+
+Tests are located in the `Tests` directory
+
+End-to-End tests often utilize the `MockServer` api to either assert against certain messages reaching the server or simply blocking until a request is sent/received
```

Comment:
```suggestion
End-to-End tests often utilize the `MockServer` API to either assert against certain messages reaching the server or simply blocking the message until a request is sent/received.
```

Reply:
I'm not sure it's accurate, but I adde "the message" to clarify what was being blocked.

Reply:
Ah, good suggestion on the clarification. In this case, we are actually blocking "the thread", so I will apply and update